testsuite/tests/shapes/recmodules.ml

```diff
@@ -43,12 +43,8 @@ and B : sig
 end = B
 [%%expect{|
 {
- ("A", module) -> {<.11>
-                   ("t", type) -> A/302<.11> . "t"[type];
-                   };
- ("B", module) -> {<.12>
-                   ("t", type) -> B/303<.12> . "t"[type];
-                   };
+ "A"[module] -> <.11>;
+ "B"[module] -> <.12>;
```

Comment:
Here too we now have a `Leaf` shape. Note that this is different from what happens in #10796 , where they are kept as variables (cf [here](https://github.com/ocaml/ocaml/pull/10796/files#diff-604a5dcfd9b59dab3398cbbfefa12dce7a5afaf8b6278482351ce6d9288685edL46)).

Reply:
(fixed, now as in #10796; I wondered why we don't get `("t", type) -> ...` anymore, this is due to the optimization on full signatures that performs an eta-reduction / avoid expanding into a new structure.)